### DISCUSSION TITLE
[Help Wanted] File storage source - S3 or server?

### DISCUSSION THREAD

**Discussion by kirtan403** [standardnotes/forum#2901] [General] [answered]
**Describe the issue**

I'm currently using the self-hosted version of the standardnotes. I wanted to know that what is the source of the files I upload. If I want to store it in the S3, is it possible? 

Or the files are stored in the server and then backed up in the S3? 

S3 backup includes files as well or only notes?

**Comment by jackyzy823** [comment]
The file storage path is under  /workspace/packages/files/uploads/ in the docker image and in self-hosted version it is mapped to a path in host machine ( see .env file FILE_UPLOAD_PATH)
you probably need docker stoarge plugin like https://hub.docker.com/r/mochoa/s3fs-volume-plugin or   https://rclone.org/docker/  to make s3 a filesystem

**Comment by kirtan403** [comment]
Thanks @jackyzy823 . My main concern is backup. Even if it stored in the local storage, I'm fine with it. But is it included in the automated s3 backups? I see a env variable `S3_BACKUP_BUCKET_NAME` . Which will include the notes backup I think. Is that include the file uploads as well or not?

**Comment by 0x111** [comment, marked as answer]
You are able to use S3 as a storage for the notes, I did set it up with the .env variables under https://github.com/standardnotes/self-hosted/blob/main/docker/files.env.sample

No s3fs or such is needed though.

**Comment by kirtan403** [comment]
Thanks @0x111 I'll try it out
